test/integration/garbagecollector/garbage_collector_test.go

```diff
@@ -258,7 +258,9 @@ func setupWithServer(t *testing.T, result *kubeapiservertesting.TestServer, work
 		t.Fatalf("failed to create garbage collector: %v", err)
 	}
 
-	stopCh := make(chan struct{})
+	// these channels need to be buffered to prevent the goroutine below from hanging indefinitely
+	// when the select statement reads something other than the one the goroutine sends on.
+	stopCh := make(chan struct{}, 1)
```

Comment:
If I understand it correctly, we send the `stop` signal by close the `stopCh` chan. 
And the `select` statement with `default` will never hanging.
```go
select {
case <-stopCh:
    return
default:
}
```

Reply:
@SataQiu fair point. If this goroutine times out there are no writers to the channel (stopCh) to cause any leak. Also, waitForStopOrTimeout() runs in a wait loop. Thank you. Closing.